packages/x-date-pickers/src/internals/demo/DemoContainer.tsx

```diff
@@ -53,21 +52,52 @@ const getSupportedSectionFromChildName = (childName: string): PickersSupportedSe
   return 'time';
 };
 
-export function DemoContainer(props: PickersGridProps) {
-  const { children } = props;
+interface DemoItemProps {
+  label?: React.ReactNode;
+  children: React.ReactNode;
+  component: string;
+}
+export function DemoItem(props: DemoItemProps) {
+  const { label, children, component } = props;
 
-  const childrenCount = React.Children.count(children);
-  const childrenTypes = new Set<PickersGridChildComponentType>();
-  const childrenSupportedSections = new Set<PickersSupportedSections>();
+  if (!component) {
```

Comment:
I'm not a fan of this extra requirement of this change.
It will pollute the demo previews a lot as well as creates two points of failure/need to edit/change in order to get "correct" layout. 🙈 

What do you think about avoiding the need of `component` prop on `DemoItem` with a tiny loss of flexibility and styling?

```diff
diff --git a/packages/x-date-pickers/src/internals/demo/DemoContainer.tsx b/packages/x-date-pickers/src/internals/demo/DemoContainer.tsx
index 78b194dea..0fd67ed39 100644
--- a/packages/x-date-pickers/src/internals/demo/DemoContainer.tsx
+++ b/packages/x-date-pickers/src/internals/demo/DemoContainer.tsx
@@ -55,35 +55,12 @@ const getSupportedSectionFromChildName = (childName: string): PickersSupportedSe
 interface DemoItemProps {
   label?: React.ReactNode;
   children: React.ReactNode;
-  component: string;
 }
 export function DemoItem(props: DemoItemProps) {
-  const { label, children, component } = props;
-
-  if (!component) {
-    throw new Error('DemoItem has to have a `component` prop defining the component it wraps.');
-  }
-
-  const childName = component;
-  const childType = getChildTypeFromChildName(childName);
-
-  let spacing: StackProps['spacing'];
-  let sx: StackProps['sx'];
-
-  if (childType === 'multi-input-range-field') {
-    spacing = 2;
-    sx = {
-      [`& .${textFieldClasses.root}`]: {
-        flexGrow: 1,
-      },
-    };
-  } else {
-    spacing = 1;
-    sx = undefined;
-  }
+  const { label, children } = props;
 
   return (
-    <Stack direction="column" alignItems="stretch" spacing={spacing} sx={sx}>
+    <Stack direction="column" alignItems="stretch" spacing={1}>
       {label && <Typography variant="body2">{label}</Typography>}
       {children}
     </Stack>
@@ -139,6 +116,15 @@ export function DemoContainer(props: DemoGridProps) {
   } else {
     sx = { [`& > .${textFieldClasses.root}`]: { minWidth: 200 } };
   }
+  // handle extra styling previously handled via `DemoItem`
+  if (childrenTypes.has('multi-input-range-field')) {
+    sx = {
+      ...sx,
+      [`& .${textFieldClasses.root}`]: {
+        flexGrow: 1,
+      },
+    };
+  }
 
   return (
     <Stack direction={direction} spacing={spacing} sx={sx}>

```